server/portal/apps/intromessages/models.py

```diff
@@ -28,3 +28,59 @@ class IntroMessages(models.Model):
     # Make each type of IntroMessage unique
     class Meta:
         unique_together = ('user', 'component',)
+
+
+class CustomMessageTemplate(models.Model):
+    """CustomMessageTemplate
+
+    Used for storing admin-controlled messages for specific components that utilize CustomMessages.
+    """
+
+    MESSAGE_TYPES = [('info', 'Info'), ('success', 'Success'), \
+                     ('warning', 'Warn'), ('error', 'Error')]
+
+    COMPONENTS = [('DASHBOARD', 'Dashboard'), ('DATA', 'Data Files'), \
+                  ('APPLICATIONS', 'Applications'), ('ALLOCATIONS', 'Allocations'), \
+                  ('HISTORY', 'History'), ('UI', 'UI'), ('ACCOUNT', 'Account')]
+
+    component = models.CharField(help_text='Component type', max_length=20, choices=COMPONENTS, default='Dashboard')
+    message_type = models.CharField(help_text='Message type', max_length=20, choices=MESSAGE_TYPES, default='info')
+    dismissible = models.BooleanField(default=True)
+    message = models.TextField(help_text='Message content', default='', blank=True)
```

Comment:
The `TextField` supports newlines, but the custom message in the frontend does not render those newlines. Do we want to allow newlines for these messages?